gh_pages/_source/session5/Simple-PCL-Interface-for-Python.rst

```diff
@@ -7,330 +7,238 @@ In this exercise, we will fill in the appropriate pieces of code to build a perc
 
-        /*
-         * SETUP RESPONSE
-         */
-          pcl::toROSMsg(*filtered_cloud, response.output_cloud);
-          response.output_cloud.header=request.input_cloud.header;
-          response.output_cloud.header.frame_id="kinect_link";
-          response.success = true;
-          return true;
-        }
 
+#. Call the service to apply a Voxel Grid filter. Find the the function for applying a voxel grid filter and insert the following code below the banner
 
-#. Within ``main``, take notice of the lines starting at 244, this is where we load the parameters used by the various filters. 
+   .. code-block:: python
+   
+        # =======================
+        # VOXEL GRID FILTER
+        # =======================
+        
+   #. We first create a service request of type FilterCloud and populate it with the necessary information:
 
-   .. code-block:: c++
+      .. code-block:: python
 
-            priv_nh_.param<double>("leaf_size", leaf_size_, 0.0f); 
-   Build the package and go into the **filter_call** package now
+           req = FilterCloud.Request()
+           req.pcdfilename = self.pcdfilename
+           req.operation = py_perception.srv.FilterCloud.Request.VOXELGRID
+           req.input_cloud = PointCloud2()
 
-#. Now that we have the framework for the filtering, open your terminal. Make sure you are in the outer ``filter_call`` directory. Create a ``scripts`` folder fo hold the python node we're creating.
+           # ERROR HANDLING
+           if req.pcdfilename == '':
+               self.get_logger().error('No file parameter found')
 
-   .. code-block:: bash
+   #. Next we can send a request to the server node and wait for a response:
 
-            mkdir scripts
-            touch scripts/filter_call.py
+      .. code-block:: python
 
-#. Copy and paste the following code at the top of ``filter_call.py`` to import necessary libraries:
+           future = self.client.call_async(req)
+           rclpy.spin_until_future_complete(self, future)
+           res_voxel = future.result()
+           if not res_voxel.success:
+               self.get_logger().error('Unsuccessful voxel grid filter operation')
 
-   .. code-block:: python
+   #. Finally, we publish our new filtered point cloud:
 
-            #!/usr/bin/env python
+      .. code-block::python
 
-            import rospy
-            import py_perception.srv
-            from sensor_msgs.msg import PointCloud2
+           self.voxel_pub.publish(res_voxel.output_cloud)
+           self.last_cloud = res_voxel
+           self.get_logger().info("published: voxel grid filter response")
 
-#. We will create an ``if`` statement that contains the ``main`` function that is called when the node is run from the command line. Paste the following after the import statements:
 
-   .. code-block:: python
+#. Before running our new node, we need to make the Python file executable. In your terminal:
```

Comment:
Colcon doesn't do this?  Is the install space script correctly marked as executable?